plugin/session_buffer.py

```diff
@@ -64,7 +64,7 @@ class SemanticTokensData:
     def __init__(self) -> None:
         self.data = []  # type: List[int]
         self.result_id = None  # type: Optional[str]
-        self.active_scopes = {}  # type: Dict[str, int]
+        self.active_scopes = []  # type: List[str]
```

Comment:
Maybe worth switching to a Set if it wouldn't cause issues? Lookup in a list is linear so not ideal with many items (not sure how many we are talking about in this case though).

Reply:
I would guess there are roughly between 0 and 20 different active token scopes. Maybe more with rust-analyzer.

Actually I wanted to use a set at first, but then decided for a list, because I thought lists would be faster due to not having the overhead from requiring items to be unique. Basically I've learned to always use "normal" arrays when performance is relevant, because their items are just consecutively in memory, and that all other fancy data structures like linked lists, etc. are just "theoretical contructs" and have no use in real world (when performance matters). But of course I'm not aware of the implementation details in Python, especially since Python's lists can contain items of different data types in a single list. Seems like according to [this SO question](https://stackoverflow.com/questions/2831212/python-sets-vs-lists) sets are faster for containment checks, and lists are faster for iterating. As both of it will be done here, it's difficult to tell which one is better. But certainly this choice is not the bottleneck in this PR :)